judo-ui-react/src/main/resources/actor/src/utilities/application/change-application.ts.hbs

```diff
@@ -1,10 +1,12 @@
 {{> fragment.header.hbs }}
 
-import { clearSecurityStorage } from '~/auth';
+import { clearSecurityStorage, getUser } from '~/auth';
 import type { HandleApplicationChange } from './interfaces';
 
 export const changeApplication: HandleApplicationChange = (applicationKey: string) => {
+  const { profile } = getUser();
+  const tokenUser = profile?.preferred_username;
   clearSecurityStorage();
   const { origin } = window.location;
-  window.location.href = origin + '/' + applicationKey;
+  window.location.href = origin + '/' + applicationKey + "?user=" + tokenUser ;
```

Comment:
Since `tokenUser` could be an email address, we might need to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent it to make sure special characters are encoded in the param. Not sure, we should check.